app/frontend/src/features/search/SearchResultsList.test.tsx

```diff
@@ -0,0 +1,136 @@
+import { getUser } from "test/serviceMockDefaults";
+import {
+  assertErrorAlert,
+  mockConsoleError,
+  MockedService,
+  wait,
+} from "test/utils";
+
+import SearchResultsList from "./SearchResultsList";
+
+const mockSearchFiltersFactory = (filters: SearchFilters = {}) => ({
+  active: filters,
+  change: jest.fn(),
+  remove: jest.fn(),
+  apply: jest.fn(),
+});
+
+const mockHandleResultClick = jest.fn();
+jest.mock("maplibre-gl");
+const mockMapRef = { current: new Map() };
+
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+
+const userSearchMock = service.search.userSearch as MockedService<
+  typeof service.search.userSearch
+>;
+
+describe("SearchResultsList", () => {
+  it("Shows a user if one is selected with no search query", async () => {
+    getUserMock.mockImplementation(getUser);
+    render(
+      <SearchResultsList
+        selectedResult={1}
+        handleResultClick={mockHandleResultClick}
+        map={mockMapRef}
+        searchFilters={mockSearchFiltersFactory()}
+      />,
+      { wrapper: hookWrapper }
+    );
+
+    expect(screen.getByRole("progressbar")).toBeVisible();
+    expect(await screen.findByRole("heading", { name: users[0].name }));
+  });
+
+  it("Is blank with no selection or query", () => {
+    render(
+      <SearchResultsList
+        selectedResult={undefined}
+        handleResultClick={mockHandleResultClick}
+        map={mockMapRef}
+        searchFilters={mockSearchFiltersFactory()}
+      />,
+      { wrapper: hookWrapper }
+    );
+    expect(screen.queryByRole("progressbar")).toBeNull();
+    expect(screen.queryAllByRole("heading")).toHaveLength(0);
+  });
+
+  it("Shows an error if selected user can't be fetched", async () => {
+    mockConsoleError();
+    getUserMock.mockRejectedValue(new Error("fetch error"));
+    render(
+      <SearchResultsList
+        selectedResult={1}
+        handleResultClick={mockHandleResultClick}
+        map={mockMapRef}
+        searchFilters={mockSearchFiltersFactory()}
+      />,
+      { wrapper: hookWrapper }
+    );
+
+    expect(screen.getByRole("progressbar")).toBeVisible();
+    await assertErrorAlert("fetch error");
+  });
+
+  describe("after searching", () => {
+    beforeEach(() => {
+      userSearchMock.mockImplementationOnce(async () => {
+        await wait(0);
+        return {
+          resultsList: [{ rank: 1, snippet: "", user: users[0] }],
+          nextPageToken: "",
+        } as UserSearchRes.AsObject;
+      });
+      render(
+        <SearchResultsList
+          handleResultClick={mockHandleResultClick}
+          map={mockMapRef}
+          searchFilters={mockSearchFiltersFactory({ query: "test query" })}
+        />,
+        { wrapper: hookWrapper }
+      );
+    });
+    it("displays the result", async () => {
+      expect(screen.getByRole("progressbar")).toBeVisible();
+      expect(await screen.findByRole("heading", { name: users[0].name }));
```

Comment:
```suggestion
      expect(await screen.findByRole("heading", { name: users[0].name })).toBeVisible();
```
Or maybe even remove the `expect(...)` around it. I guess it's ok if it's only used as a handle to wait for something first before proceeding with the next assertion.